File: GoldmanSachs/JavaScript/PrintAnagramsTogether.js
// I do not yet understand why I needed to sort the final_list here and not in cpp but I will find out.
// I understand. It has nothing to do with my code, and everything to do with the crappy build code the GFG developers/contributors have injected into their website that work differently for each language and creates complications for the innocent user.

class Solution {
    Anagrams(n, string_list){
        let final_list = [];
        let dict = {};
        
        string_list.forEach((v, i, arr) => {
            let key = v.split("").sort().join("");
            if (!dict[key]) {
                dict[key] = [];
            }
            dict[key].push(v);
        });
        
        for (const prop in dict) {
            final_list.push(dict[prop]);
        }
        
        return final_list.sort();
    }
}

File: GoldmanSachs/C#/PrintAnagramsTogether.cs
// I do not currently know whether this code will work or not because the GFG people have a bug on their website (shameless!)
// I have to wait till they fix it. (I know I can test myself, but I don't wanna.)

class Solution
{
    public List<List<string>> Anagrams(List<string> string_list)
    {
        List<List<string>> finalList = new List<List<string>>();
        Dictionary<string, List<string>> dict = new Dictionary<string, List<string>>();
        
        foreach (string str in string_list) {
            char[] charArr = str.ToCharArray();
            Array.Sort(charArr);
            string newStr = string.Join("", charArr);
            dict[newStr].Add(str);
        }
        
        foreach (KeyValuePair<string, List<string>> pair in dict) {
            finalList.Add(pair.Value);
        }
        
        return finalList;
    }
}
